src/Fractionalizer.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import { ERC1155Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC1155/ERC1155Upgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { IERC1155 } from "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import { ERC1155ReceiverUpgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC1155/utils/ERC1155ReceiverUpgradeable.sol";
+import { CountersUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+
+import { ERC1155Supply } from "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IPNFT } from "./IPNFT.sol";
+import { SchmackoSwap, ListingState } from "./SchmackoSwap.sol";
+
+/// @title Fractionalizer
+/// @author molecule.to
+/// @notice
+contract Fractionalizer is ERC1155Upgradeable, UUPSUpgradeable, OwnableUpgradeable {
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+
+    struct Withdrawal {
+        uint256 listingId;
+        IERC20 paymentToken;
+        uint256 tokenAmount;
+    }
+
+    struct Fractions {
+        address collection;
+        uint256 nftId;
+        address lockingAccount;
+        uint256 issued;
+        uint256 stillAvailable;
```

Comment:
I would probably rename to `locked` - otherwise 'Available' might be misinterpreted as liquid